sympy/functions/special/error_functions.py

```diff
@@ -1434,26 +1434,52 @@ def _eval_rewrite_as_Si(self, nu, z, **kwargs):
     _eval_rewrite_as_Chi = _eval_rewrite_as_Si
     _eval_rewrite_as_Shi = _eval_rewrite_as_Si
 
+    def _eval_rewrite_as_tractable(self, nu, z, limitvar=None, **kwargs):
+        return exp(-z)*_expint(nu, z)
+
+    def _eval_as_leading_term(self, x, logx, cdir):
+        if not self.args[0].has(x):
+            try:
+                _, ex = self.args[1].leadterm(x)
+            except (ValueError, NotImplementedError):
+                return self
```

Comment:
How necessary are these try-except blocks here ?

They might be important but I don't see any tests being added for this case. Maybe you can add some ?